components/uaepass-authenticator/src/main/java/org/wso2/carbon/identity/uaepass/authenticator/UAEPassAuthenticator.java

```diff
@@ -0,0 +1,1109 @@
+
+        try {
+            OAuthAuthzResponse authzResponse = OAuthAuthzResponse.oauthCodeAuthzResponse(request);
+            OAuthClientRequest accessTokenRequest = getAccessTokenRequest(context, authzResponse);
+            OAuthClient oAuthClient = new OAuthClient(new URLConnectionClient());
+            OAuthClientResponse oAuthResponse = getOauthResponse(oAuthClient, accessTokenRequest);
+
+            String accessToken = oAuthResponse.getParam(UAEPassAuthenticatorConstants.UAE.ACCESS_TOKEN);
+            if (StringUtils.isBlank(accessToken)) {
+                if(LOG.isDebugEnabled()){
+                    LOG.debug("Access Token is empty");
+                }
+                throw new AuthenticationFailedException("Access token is empty");
+            }
+
+            String idToken = oAuthResponse.getParam(UAEPassAuthenticatorConstants.UAE.ID_TOKEN);
+
+            AuthenticatedUser authenticatedUser;
+            Map<String, Object> jsonClaimMap;
+            Map<ClaimMapping, String> claims = new HashMap<>();
+
+            if (StringUtils.isNotBlank(idToken)) {
+                if(LOG.isDebugEnabled()){
+                    LOG.debug("Id token available from UAEPass");
+                }
+                context.setProperty(UAEPassAuthenticatorConstants.UAE.ACCESS_TOKEN, accessToken);
+                jsonClaimMap = getIdTokenClaims(context, idToken);
+            } else {
+                jsonClaimMap = getUserInfoUserAttributes(oAuthResponse, context);
+            }
+
+            String authenticatedUserId = getAuthenticatedUserId(context, jsonClaimMap);
+            String attributeSeparator = getMultiAttributeSeparator(context, authenticatedUserId);
+
+            jsonClaimMap.entrySet().stream()
+                    .filter(entry -> !ArrayUtils.contains(UAEPassAuthenticatorConstants.UAEPassRuntimeConstants.
+                            NON_USER_ATTRIBUTES, entry.getKey()))
+                    .forEach(entry -> buildClaimMappings(claims, entry, attributeSeparator));
+
+            if (StringUtils.isBlank(authenticatedUserId)) {
+                throw new AuthenticationFailedException("Cannot find the userId from the id_token sent "
+                        + "by the federated IDP.");
+            }
+            authenticatedUser = AuthenticatedUser.createFederateAuthenticatedUserFromSubjectIdentifier
+                    (authenticatedUserId);
+            authenticatedUser.setUserAttributes(claims);
+            context.setSubject(authenticatedUser);
+
+        } catch (OAuthProblemException e) {
+            LOG.error("Authentication client response failure",e);
+        } catch (UAEPassUserInfoFailedException e){
+            throw new AuthenticationFailedException("Unable to build Access token request", e);
+        } catch (OAuthSystemException e) {
+            LOG.error("Authorize response Initiation Failed : ",e);
```

Comment:
```suggestion
            LOG.error("Authorize response Initiation Failed : ", e);
```
Check formatting in other places as well